hrm-service/src/case/case.ts

```diff
@@ -106,33 +194,97 @@ const generateLegacyNotesFromCounsellorNotes = caseFromDb => {
   return caseFromDb;
 };
 
-export const createCase = async (body, accountSid, workerSid): Promise<CaseRecord> => {
-  const migratedBody = migrateAddedLegacyNotesToCounsellorNotes(
-    fixLegacyReferrals(body, workerSid),
+/**
+ * Converts a case passed in from the API to a case record ready to write to the DB
+ * If an 'original' case is passed, it is assumed this is an update of an existing case
+ * Data from the 'original will be used to inject data that might be missing from the update if it is using a legacy format
+ * @param inputCase
+ * @param workerSid
+ * @param original
+ */
+const caseToCaseRecord = (
+  inputCase: Partial<Case>,
+  workerSid: string,
+  original?: Case,
+): Partial<NewCaseRecord> => {
+  const migratedCase = migrateAddedLegacyNotesToCounsellorNotes(
+    fixLegacyReferrals(inputCase, workerSid, original),
     workerSid,
+    original,
+  );
+  const info = migratedCase.info ?? {};
+  const caseSections: CaseSectionRecord[] = Object.entries(WELL_KNOWN_CASE_SECTION_NAMES).flatMap(
+    ([sectionName, { getSectionSpecificData, sectionTypeName }]) =>
+      (info[sectionName] ?? []).map(section => {
+        const caseSectionRecordToUpsert: CaseSectionRecord = {
+          caseId: migratedCase.id,
+          sectionType: sectionTypeName,
+          sectionId: section.id ?? randomUUID(),
```

Comment:
Can we still have an empty `sectionId` even after the migration on https://github.com/techmatters/hrm/pull/162 is run? Or this is just a sanity check?

Reply:
Clients could still feasibly provide new cases without IDs specified - I opted to create one rather than failing the operation, just for the case of old clients, but we could be more strict in the future